pkg/inputs/list_select/model.go

```diff
@@ -0,0 +1,28 @@
+package listselect
+
+import "github.com/charmbracelet/bubbles/list"
+
+type SelectInput[T any] struct {
```

Comment:
the generic implementation is nice, but I wonder if it's not too generic in a way that it causes too much overhead and complexity. let's discuss